### PR TITLE
[libc++] fix `array<T, 0>` under `USE_WRAP_ITER`

### PR DESCRIPTION
**Pull request by strega-nil**
`array<T, 0>::iterator` was always a pointer even when `_LIBCXX_ABI_USE_WRAP_ITER_IN_STD_ARRAY` was defined. This patch fixes that minor bug.

Discovered as part of [#100603][].

[#100603]: https://github.com/llvm/llvm-project/pull/100603

### CODE REVIEW

File: libcxx/include/array [hunk 1]
```diff
@@ -270,18 +270,23 @@ struct _LIBCPP_TEMPLATE_VIS array {
...
+#else
+  using iterator       = pointer;
+  using const_iterator = const_pointer;
+#endif
+  using size_type              = size_t;
+  using difference_type        = ptrdiff_t;
+  using reverse_iterator       = std::reverse_iterator<iterator>;
+  using const_reverse_iterator = std::reverse_iterator<const_iterator>;
 
   typedef __conditional_t<is_const<_Tp>::value, const __empty, __empty> _EmptyType;
```
Comment (Zingam) [inline libcxx/include/array:291]: For consistency's sake maybe also change this typedef to using as a drive-by improvement?

Comment (strega-nil) [reply libcxx/include/array:291]: can do.

**Review by ldionne — APPROVED** [review]
Is that the reason we were introducing new broken test code without noticing?

Either way, thanks for fixing.

**Comment by strega-nil** [comment]
@ldionne exactly :)
